### ISSUE TITLE
Eliminate All Generic Exception Handling

### ISSUE THREAD

**Issue by adamkrellenstein** [CounterpartyXCP/counterparty-core#1381]
<img width="983" alt="Screenshot 2024-02-01 at 12 06 36 PM" src="https://github.com/CounterpartyXCP/counterparty-lib/assets/6189829/21342ec4-89fd-4190-9239-ccb9ef0708d3">

**Comment by ouziel-slama** [comment]
In this generic except we just cast any error in `ParseTransactionError()`.. it's only cosmetic we could just delete these two lines.. the main thing is, here, if there is an error while parsing a transaction the server crashes..and that's what we want because that should never happen..  I think we should just leave it like that.. on the other hand we have to make sure that there is no generic Except during the parsing of transactions which would make this Except useless.. normally we have already removed those that were there..

**Comment by adamkrellenstein** [comment]
Sorry yes that's what I meant for this issue... remove *all* generic exception handling!

Maybe we should do that as part of a protocol change though?

**Comment by ouziel-slama** [comment]
yeah I think so! There are a few left in `unpack` functions but I looked around and a priori nothing dangerous.
